### ISSUE TITLE
[Bug]: After installing the window, the configuration linked to the model in the gpustack, but the chat box returned an empty response.

### ISSUE THREAD

**Issue by cuihangrui** [openclaw/openclaw#42930]
### Bug type

Behavior bug (incorrect output/state without crash)

### Summary

I installed the latest openclaw 2026.3.8 and configured gpustack 2.1.0. Through the chat function on the openclaw page, there were no errors on the backend, and the log call of gpustack was also successful.

gpustack  logs   =====

srv  params_from_: Chat format: peg-native
slot get_availabl: id  1 | task -1 | selected slot by LCP similarity, sim_best = 0.993 (> 0.100 thold), f_keep = 0.997
slot launch_slot_: id  1 | task -1 | sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist 
slot launch_slot_: id  1 | task 319 | processing task, is_child = 0
slot update_slots: id  1 | task 319 | new prompt, n_ctx_slot = 32768, n_keep = 0, task.n_tokens = 9619
slot update_slots: id  1 | task 319 | n_tokens = 9552, memory_seq_rm [9552, end)
slot init_sampler: id  1 | task 319 | init sampler, took 4.38 ms, tokens: text = 9619, total = 9619
slot update_slots: id  1 | task 319 | prompt processing done, n_tokens = 9619, batch.n_tokens = 67
srv  log_server_r: done request: POST /v1/responses 192.168.100.97 200
slot print_timing: id  1 | task 319 | 
prompt eval time =      65.37 ms /    67 tokens (    0.98 ms per token,  1024.92 tokens per second)
       eval time =     188.10 ms /    24 tokens (    7.84 ms per token,   127.59 tokens per second)
      total time =     253.47 ms /    91 tokens
slot      release: id  1 | task 319 | stop processing: n_tokens = 9642, truncated = 0
srv  update_slots: all slots are idle
===
My oepclaw.json is as follows

<img width="1660" height="1150" alt="Image" src="https://github.com/user-attachments/assets/ad00e1f2-3556-46aa-a9b0-b0ef5bf35b6f" />

<img width="1357" height="1193" alt="Image" src="https://github.com/user-attachments/assets/20404489-bbdb-44d9-827e-46d8f5835b37" />

<img width="2552" height="1344" alt="Image" src="https://github.com/user-attachments/assets/fc315cdd-b833-485d-bffd-ea5c754d31ea" />

<img width="1535" height="808" alt="Image" src="https://github.com/user-attachments/assets/260e987b-9324-4102-bd91-d4c84b37da81" />

### Steps to reproduce

Help me check if my screenshots, content and OpenClaw configuration file are all correct.

### Expected behavior

Help me check if my screenshots, content and OpenClaw configuration file are all correct.

### Actual behavior

The reply on the chat page is empty.

logs   ====
🦞 OpenClaw 2026.3.8 (3caab92) — Your task has been queued; your dignity has been deprecated.

(node:24832) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
14:57:37 [gateway] feishu_doc: Registered feishu_doc, feishu_app_scopes
14:57:37 [gateway] feishu_chat: Registered feishu_chat tool
14:57:37 [gateway] feishu_wiki: Registered feishu_wiki tool
14:57:37 [gateway] feishu_drive: Registered feishu_drive tool
14:57:37 [gateway] feishu_bitable: Registered bitable tools
14:57:37 [canvas] host mounted at http://127.0.0.1:18789/__openclaw__/canvas/ (root C:\Users\xiqi\.openclaw\canvas)
14:57:37 [heartbeat] started
14:57:37 [health-monitor] started (interval: 300s, startup-grace: 60s, channel-connect-grace: 120s)
14:57:37 [gateway] agent model: vllm/qwen3-instruct-2507
14:57:37 [gateway] listening on ws://127.0.0.1:18789, ws://[::1]:18789 (PID 24832)
14:57:37 [gateway] log file: C:\Users\xiqi\AppData\Local\Temp\openclaw\openclaw-2026-03-11.log
14:57:37 [info]: [ 'client ready' ]
14:57:37 [browser/server] Browser control listening on http://127.0.0.1:18791/ (auth=token)
14:57:37 [feishu] starting feishu[default] (mode: websocket)
14:57:38 [ws] webchat connected conn=d8669e78-04dd-407c-9674-0885e30c4f25 remote=127.0.0.1 client=openclaw-control-ui webchat v2026.3.8
14:57:38 [feishu] feishu[default]: bot open_id resolved: ou_ad4d874d274ceb06c72dbb01a7273046
14:57:38 [info]: [ 'event-dispatch is ready' ]
14:57:38 [feishu] feishu[default]: starting WebSocket connection...
14:57:38 [info]: [
  '[ws]',
  'receive events or callbacks through persistent connection only available in self-build & Feishu app, Configured in:\n' +
    '        Developer Console(开发者后台) \n' +
    '          ->\n' +
    '        Events and Callbacks(事件与回调)\n' +
    '          -> \n' +
    '        Mode of event/callback subscription(订阅方式)\n' +
    '          -> \n' +
    '        Receive events/callbacks through persistent connection(使用 长连接 接收事件/回调)'
]
14:57:38 [feishu] feishu[default]: WebSocket client started
14:57:38 [info]: [ '[ws]', 'ws client ready' ]
15:03:08 [ws] webchat disconnected code=1001 reason=n/a conn=d8669e78-04dd-407c-9674-0885e30c4f25
15:03:08 [ws] webchat connected conn=63716f29-6d71-4fe2-8f90-54e6b2611430 remote=127.0.0.1 client=openclaw-control-ui webchat v2026.3.8

### OpenClaw version

2026.3.8

### Operating system

window11 

### Install method

iwr -useb https://openclaw.ai/install.ps1 | iex

### Model

qwen3-instruct-2507

### Provider / routing chain

gpustack

### Logs, screenshots, and evidence

```shell

```

**Comment by steipete** [comment]
Closing this as implemented after Codex review.

Current `main` already handles zero-text assistant turns by converting them into an explicit incomplete-turn error or retry path instead of letting WebChat end with an empty assistant response, and the vLLM docs now point OpenAI-compatible local servers at `openai-completions`/`/v1/chat/completions`.

What I checked:

- **Empty-response turns are now detected and surfaced as an explicit user-visible error:** `resolveIncompleteTurnPayloadText()` treats zero-payload, zero-text assistant turns as `emptyResponseAssistant` and returns `⚠️ Agent couldn't generate a response. Please try again.` instead of silently succeeding. ([`src/agents/pi-embedded-runner/run/incomplete-turn.ts:160`](https://github.com/openclaw/openclaw/blob/a29233c9af0c/src/agents/pi-embedded-runner/run/incomplete-turn.ts#L160), [a29233c9af0c](https://github.com/openclaw/openclaw/commit/a29233c9af0c))
- **Embedded runner now retries/surfaces empty responses instead of leaving a silent gap:** `runEmbeddedPiAgent()` calls `resolveEmptyResponseRetryInstruction()`, logs `empty response detected`, retries once, and if still empty logs `empty response retries exhausted` before surfacing the incomplete-turn error. ([`src/agents/pi-embedded-runner/run.ts:1804`](https://github.com/openclaw/openclaw/blob/a29233c9af0c/src/agents/pi-embedded-runner/run.ts#L1804), [a29233c9af0c](https://github.com/openclaw/openclaw/commit/a29233c9af0c))
- **Regression tests cover the exact empty-response symptom:** `run.incomplete-turn.test.ts` asserts that a zero-text assistant turn is retried and then surfaced as an error after exhaustion; `run.empty-error-retry.test.ts` adds model-agnostic no-visible-text retry coverage for empty error turns. ([`src/agents/pi-embedded-runner/run.incomplete-turn.test.ts:398`](https://github.com/openclaw/openclaw/blob/a29233c9af0c/src/agents/pi-embedded-runner/run.incomplete-turn.test.ts#L398), [a29233c9af0c](https://github.com/openclaw/openclaw/commit/a29233c9af0c))
- **Current vLLM docs no longer steer OpenAI-compatible local servers to Responses mode:** The vLLM provider guide says OpenClaw connects to vLLM with `openai-completions`, shows `/v1/chat/completions`, and uses `api: "openai-completions"` in the config example. Public docs: [`docs/providers/vllm.md`](https://docs.openclaw.ai/providers/vllm). ([`docs/providers/vllm.md:9`](https://docs.openclaw.ai/providers/vllm), [564f820efa2b](https://github.com/openclaw/openclaw/commit/564f820efa2b))

So I’m closing this as already implemented rather than keeping a duplicate issue open.

Review notes: reviewed against [564f820efa2b](https://github.com/openclaw/openclaw/commit/564f820efa2bb81ae49b8f55a93e41492ad31917); fix evidence: commit [a29233c9af0c](https://github.com/openclaw/openclaw/commit/a29233c9af0c03e906b5a1e3dceb927119eafe04).
